modules/apps/frontend-view-state/frontend-view-state-service/src/main/java/com/liferay/frontend/view/state/internal/jaxrs/application/FrontendViewStateApplication.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.view.state.internal.jaxrs.application;
+
+import com.liferay.frontend.view.state.model.FVSActiveEntry;
+import com.liferay.frontend.view.state.model.FVSEntry;
+import com.liferay.frontend.view.state.service.FVSActiveEntryLocalService;
+import com.liferay.frontend.view.state.service.FVSEntryLocalService;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.User;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.jaxrs.whiteboard.JaxrsWhiteboardConstants;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(
+	property = {
+		JaxrsWhiteboardConstants.JAX_RS_APPLICATION_BASE + "=/frontend-view-state/app",
+		JaxrsWhiteboardConstants.JAX_RS_NAME + "=Liferay.Frontend.View.State",
+		"auth.verifier.auth.verifier.PortalSessionAuthVerifier.urls.includes=/*",
+		"auth.verifier.guest.allowed=true", "liferay.oauth2=false"
+	},
+	service = Application.class
+)
+public class FrontendViewStateApplication {
+
+	@GET
+	@Path("/active-view/{clayDataSetDisplayId}")
```

Comment:
hopefully Brian won't complain because you are not using REST Builder. But it's probable that he'll complain because the path is not `"/active-view/{activeViewId}"`. Given that we are generalizing this to be used for other view componentes, I would rename it.

Reply:
Not sure, because he changed `dataSetDisplayId` to `clayDataSetDisplayId` in a previous commit, which was quite unexpected, given the reasons you mention: generalization.

But let's do as you say and see what happens. I prefer using generic names too. :+1: